src/v/cloud_topics/level_one/compaction/source.cc

```diff
@@ -32,14 +32,17 @@ namespace {
 
 class map_building_reducer {
 public:
-    struct map_building_result {
-        std::optional<kafka::offset> min_offset_indexed{std::nullopt};
-        std::optional<kafka::offset> max_offset_indexed{std::nullopt};
-        bool map_is_full{false};
+    struct return_t {
+        bool map_is_full;
+        std::optional<model::offset> max_indexed_offset;
```

Comment:
I'm a little nervous of this returning a single offset, vs returning the offset interval set representing the offsets added to the map